standard/stream_links.lua

```diff
@@ -51,4 +55,54 @@ function StreamLinks.readCountdownStreams(args)
 	return stream
 end
 
+--[[
+Resolves the value of a stream given the platform
+]]
+function StreamLinks.resolve(platformName, streamValue)
+	local streamLink = mw.ext.StreamPage.resolve_stream(platformName, streamValue)
+
+	return string.gsub(streamLink, 'Special:Stream/' .. platformName, '')
+end
+
+--[[
+Extracts the streaming platform args from an argument table or a nested stream table inside the arguments table.
+Uses variable fallbacks and resolves stream redirects.
+]]
+function StreamLinks.processStreams(args)
+	local streams = {}
+	if type(args.stream) == 'table' then
+		streams = args.stream
+	end
+
+	for _, platformName in pairs(StreamLinks.countdownPlatformNames) do
+		-- stream has no platform and might be a table or a string
+		if platformName == 'stream' then
+			local streamValue = streams.stream
+			if String.isEmpty(streamValue) and type(args.stream) == 'string' then
+				streamValue = args.stream
+			end
+			if String.isEmpty(streamValue) then
+				streamValue = Variables.varDefault(platformName)
+			end
+			if String.isNotEmpty(streamValue) then
+				streams.stream = streamValue
+			end
+		else
+			local streamValue = Logic.emptyOr(streams[platformName] or args[platformName], Variables.varDefault(platformName))
+
+			-- twitch2 is not a platform but uses the twitch platform instead
+			local lookUpPlatform = platformName
+			if platformName == 'twitch2' then
+				lookUpPlatform = 'twitch'
```

Comment:
Setup a generic data structure for this

Reply:
you mean something like the following?
```lua
			local lookUpPlatform = string.gsub(platformName, '%d$', '')
```
so far only twitch has the additional supported param (it is limited by the table that holds the supported values and by module:Countdown)
so that might be a bit overkill

Reply:
was more thinking like among the lines of
```lua
local _REAL_PLATFORM = { --name tbd
    twitch2 = twitch,
}```
with 
```lua
			local lookUpPlatform = _REAL_PLATFORM[platformName] or platformName
```

Reply:
done, but made the table accessible from outside the module in case it is needed by other modules